app/src/main/java/com/navapp/navigation/route/RouteOptimizer.java

```diff
@@ -0,0 +1,11 @@
+package com.navapp.navigation.route;
+
+import com.navapp.navigation.destination.Destination;
+import com.navapp.navigation.destination.data.Address;
+
+import java.util.Collection;
+import java.util.Deque;
+
+public interface RouteOptimizer {
```

Comment:
Maybe a `Queue`? makes more since

Reply:
sense*

Reply:
yeah...

Reply:
You didn't change it though

Reply:
omg did it all changes on the wrong branch